qiskit/visualization/matplotlib.py

```diff
@@ -863,11 +863,19 @@ def _get_colors(self, node):
 
     def _condition(self, node, cond_xy):
         """Add a conditional to a gate"""
+        cond_is_bit = bool(isinstance(node.op.condition[0], Clbit))
         mask = 0
         qubit_b = min(self._data[node]["q_xy"], key=lambda xy: xy[1])
         for index, cbit in enumerate(self._clbit):
-            if self._bit_locations[cbit]["register"] == node.op.condition[0]:
-                mask |= 1 << index
+            if cond_is_bit:
+                for index, cbit in enumerate(self._clbit):
+                    if cbit == node.op.condition[0]:
```

Comment:
I think you have an extra `for` loop at line 869.

Reply:
Ah. Thanks for pointing to out. Fixed it in https://github.com/Qiskit/qiskit-terra/pull/6259/commits/1d9c58fdb9611478fb3670798be43c05a503d499 .